windows/src/engine/keyman32/Keyman32.cpp

```diff
@@ -565,18 +565,28 @@ extern "C" BOOL  _declspec(dllexport) WINAPI Keyman_ForceKeyboard(PCSTR FileName
 	_td->lpActiveKeyboard->IMDLLs = NULL;
   _td->lpActiveKeyboard->KeyboardOptions = NULL;*/
   _splitpath_s(FileName, NULL, 0, NULL, 0, _td->lpActiveKeyboard->Name, sizeof(_td->lpActiveKeyboard->Name), NULL, 0);
-
-	if(LoadKeyboard(_td->ForceFileName, &_td->lpActiveKeyboard->Keyboard))
-	{
-		SendDebugMessageFormat(0,sdmGlobal,0,"Keyman_ForceKeyboard: %s OK", FileName);
-		ResetCapsLock();
-		LoadDLLs(_td->lpActiveKeyboard);
-		ActivateDLLs(_td->lpActiveKeyboard);
-    LoadKeyboardOptions(_td->lpActiveKeyboard);   // I2437 - Crash unloading keyboard due to keyboard options not set
-    RefreshPreservedKeys(TRUE);
-		return TRUE;
-	}
-
+  // TODO: 5442 - remove if/ else as there will no longer be the old LoadKeyboard option
+  if (Globals::get_CoreIntegration()) {
+    PWCHAR keyboardPath   = strtowstr(_td->ForceFileName);
+    km_kbp_status_codes err_code = (km_kbp_status_codes)km_kbp_keyboard_load(keyboardPath, &_td->lpActiveKeyboard->coreKeyboard);
+    if (err_code != KM_KBP_STATUS_OK) {
+      delete keyboardPath;
+      return FALSE;
+    }
+    delete keyboardPath;
+    SendDebugMessageFormat(0, sdmGlobal, 0, "Keyman_ForceKeyboard Core: %s OK", FileName); // TODO: 5442 - remove word Core 
+    return TRUE;
+  } else {
+    if (LoadKeyboard(_td->ForceFileName, &_td->lpActiveKeyboard->Keyboard)) {
+      SendDebugMessageFormat(0, sdmGlobal, 0, "Keyman_ForceKeyboard: %s OK", FileName);
+      ResetCapsLock();
+      LoadDLLs(_td->lpActiveKeyboard);
+      ActivateDLLs(_td->lpActiveKeyboard);
+      LoadKeyboardOptions(_td->lpActiveKeyboard);  // I2437 - Crash unloading keyboard due to keyboard options not set
```

Comment:
We will need to do something corresponding to each of these four lines in the core code path.

Reply:
Yes can we do this in separate PRs?. See the Limitations TODO in issue #5011
Limitations - Now a TODO list
There are limitations that will not be addressed in phase one #5443 of the integration of the core processor these are:

- [ ] 
- [ ]  Integrate core feat(windows): Keyman Core integration 🥑 #5443
- [ ]  Support for IMX DLLs - This will be blocked out by the registry flag for the initial implementation.
- [ ]  Full support for Caps Lock system stores (shift frees caps, caps on only, caps always off)
- [ ]  Saving/restoring keyboard options either side of processing a key stroke.